### PR TITLE
S1944: Add support for deconstruction

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
The test was missing a concrete implementation. I corrected it and the diagnostic turned up.

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/InvalidCastToInterface.CSharp10.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ﻿public interface IMyInterface { }
 
+public class SomeImpl : IMyInterface { }
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/InvalidCastToInterface.CSharp10.cs:3]: Ok, definitely there is nothing to be done from the point of our current, ongoing sprint. But shouldn`t the rule be updated? I do not really understand WHY is a concrete implementation needed in order to have an issue reported?

Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/InvalidCastToInterface.CSharp10.cs:3]: I can tell why that is, but the behavior is explicitly called out in the RSpec https://sonarsource.github.io/rspec/#/rspec/S1944/csharp

> Exceptions
> No issue is reported if the interface has no implementing class in the assembly.

I agree with you that this limitation is nonsensical. Should we create an issue?

Comment (csaba-sagi-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/InvalidCastToInterface.CSharp10.cs:3]: I suggest to start a discussion on this topic on our slack channel, maybe others see a reason why this exception exists. If not, we can open an issue to fix it in the future.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**